src/wxProfilerGUI/threadlist.cpp

```diff
@@ -228,11 +234,28 @@ void ThreadList::updateSorting()
 	fillList();
 }
 
-int ThreadList::getNumDisplayedThreads() {
+int ThreadList::getNumDisplayedThreads()
+{
 	int numThreads = (int) threads.size();
 	return numThreads < MAX_NUM_DISPLAYED_THREADS ? numThreads : MAX_NUM_DISPLAYED_THREADS;
 }
 
+std::string formatDuration(__int64 durationMs)
+{
+	char str[32];
+	if (durationMs >= 0) {
+		__int64 upTimeS = durationMs / 1000;
+		int h = (int) (upTimeS / 3600);
+		int m = (int) ((upTimeS % 3600) / 60);
+		int s = (int) (upTimeS % 60);
+		sprintf(str, "%02d:%02d:%02d", h, m, s);
+	}
+	else {
+		strcpy(str, "-");
+	}
+	return str;

```

Comment:
I tried to closely follow the programming style used by the rest of the code. Fixed-length arrays were used there for formatting. But I agree with your assessment and would not write code like this in my own codebase :).